flux-desugar/src/zip_resolver.rs

```diff
@@ -65,7 +64,7 @@ fn zip_arg(arg: Arg, rust_ty: &rustc_ty::Ty) -> Arg<Res> {
     }
 }
 
-fn zip_ty(ty: Ty, rust_ty: &rustc_ty::Ty) -> Ty<Res> {
+pub fn zip_ty(ty: Ty, rust_ty: &rustc_ty::Ty) -> Ty<Res> {
```

Comment:
Is the `pub` really needed? I can't find any uses outside the module

```suggestion
fn zip_ty(ty: Ty, rust_ty: &rustc_ty::Ty) -> Ty<Res> {
```